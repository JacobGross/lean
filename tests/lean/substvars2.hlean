inductive list (A : Type) :=
| nil  : list A
| cons : A → list A → list A

open nat prod

example (A B : Type) (d c : nat) (h₀ : c = 0) (a : A) (b : list B) (h₁ : A = list B) (h₂ : eq.rec_on h₁ a = @list.nil B) (h₃ : d = c) (h₄ : d + 1 = d + 2)
        : b = eq.rec_on h₁ a × c = 1:=
begin
  substvars,
  state,
  injection h₄,
  contradiction
end
