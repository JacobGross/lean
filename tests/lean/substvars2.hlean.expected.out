substvars2.hlean:11:2: proof state
A B : Type,
a : A,
b : list B,
h₁ : A = list B,
h₂ : eq.rec_on h₁ a = list.nil B,
h₄ : 0 + 1 = 0 + 2
⊢ b = eq.rec_on h₁ a × 0 = 1
